sasl/base/src/main/java/org/wildfly/security/sasl/util/AbstractSaslParticipant.java

```diff
@@ -206,7 +206,12 @@ public void setNegotiationState(final int newState) {
      */
     public void negotiationComplete() {
         state = COMPLETE_STATE;
-        log.tracef("SASL Negotiation Completed");
+        final String msg = "SASL Negotiation Completed";
+        if (log != null) {
+            log.tracef(msg);
+        } else {
+            legacyLog.tracef(msg);
+        }
```

Comment:
Just a minor comment, this could be simplified to:

`log != null ? log.tracef(msg) : legacyLog.tracef(msg);`